.github/actions/setup-externals/action.yaml

```diff
@@ -0,0 +1,134 @@
+name: 'setup-externals'
+description: 'Setup externals such as the Intel compiler, MPI libraries.'
+inputs:
+  compiler:
+    description: 'Compiler (available options gcc@9, gcc@10, apple-clang, intel)'
+    required: true
+    default: 'gcc@9'
+  mpi:
+    description: 'Which MPI flavor (openmpi, mpich, intel-oneapi-mpi)'
+    required: false
+    default: 'mpich@4'
+
+runs:
+  using: "composite"
+
+  steps:
+
+  - name: os-setup
+    uses: ./.github/actions/setup-os
+
+  - name: cache-opt-intel
+    id: cache-opt-intel
+    uses: actions/cache@v2
+    if: runner.os == 'Linux'
+    with:
+      path: /opt/intel
+      key: opt-intel-${{ inputs.mpi }}-${{ inputs.compiler }}.${{ runner.os }}3
+
+  - name: install-opt-intel
+    shell: bash
+    env:
+      OMP_NUM_THREADS: 4
+    if: steps.cache-opt-intel.outputs.cache-hit != 'true' && runner.os == 'Linux'
+    run: |
+      if [[ "$RUNNER_OS" == "Linux" ]]; then
```

Comment:
This line (and 47-49) aren't really necessary. I added the runner condition in line 33 afterwards and didn't take 35 and 47-49 out. Can be done without having to rerun the CI tests.